lib/run_loop/core.rb

```diff
@@ -68,7 +68,8 @@ def self.run_with_options(options)
       instruments = options[:instruments] || RunLoop::Instruments.new
 
       # Find the Device under test, the App under test, UIA strategy, and reset options
-      device = RunLoop::Device.detect_device(options, xcode, simctl, instruments)
+      # Experimental selected device detection
+      device = RunLoop::DetectAUT::Xcode.detect_selected_device || RunLoop::Device.detect_device(options, xcode, simctl, instruments)
```

Comment:
RunLoop::Device.detect_device should do all the work; it should call out to `detect_selected_device`.